Lab2/app/src/main/java/nm/lab2/DatabaseFragment.java

```diff
@@ -0,0 +1,131 @@
+                             Bundle savedInstanceState) {
+        View rootView = inflater.inflate(R.layout.fragment_database, container, false);
+
+        // define parts of interface
+        Button searchFragmentButton = (Button) rootView.findViewById(R.id.search_fragment_button);
+        searchFragmentButton.setOnClickListener(new ChangeBackFragmentListener());
+        Button next = (Button) rootView.findViewById(R.id.next_button_database);
+        next.setOnClickListener(new NextListener());
+        Button previous = (Button) rootView.findViewById(R.id.previous_button_database);
+        previous.setOnClickListener(new PreviousListener());
+        Button delete = (Button) rootView.findViewById(R.id.delete_button);
+        delete.setOnClickListener(new DeleteListener());
+        webView = (WebView) rootView.findViewById(R.id.database_web_view);
+
+        //retrieve from database
+        getSavedImages();
+        //put first image in webView if it exists
+        if (num_images != 0){
+            webView.loadUrl(images.get(0));
+        } else{
+            Toast.makeText(getActivity().getBaseContext(), "No images have been saved",
+                    Toast.LENGTH_LONG).show();
+        }
+
+        return rootView;
+    }
+
+    //get images from database and save them into arraylist "images"
+    public void getSavedImages() {
+        mDbHelper = new FeedReaderDbHelper(getActivity()); //initialize database
+        images = mDbHelper.readDatabase();
+        num_images = images.size();
+    }
+
+    //change back to search fragment
+    public class ChangeBackFragmentListener implements View.OnClickListener {
+        @Override
+        public void onClick(View v) {
+            getActivity().getSupportFragmentManager().beginTransaction()
+                    .replace(R.id.container, new SearchFragment())
+                    .commit();
+        }
+    }
+
+    //next button listener
+    public class NextListener implements View.OnClickListener {
+        @Override
+        public void onClick(View v) {
+            if(current_image < num_images-1){
+                //if there are still images
+                current_image += 1;
+                webView.loadUrl(images.get(current_image));
+            }else{
+                //if looking at last image
+                Toast.makeText(getActivity().getBaseContext(), "No more images to view",
+                        Toast.LENGTH_SHORT).show();
+            }
+        }
+    }
+
+    //previous button listener
+    public class PreviousListener implements View.OnClickListener {
+
+        @Override
+        public void onClick(View v) {
+            if(current_image > 0){
+                //if you aren't on the first image
+                current_image -= 1;
+                webView.loadUrl(images.get(current_image));
+            }else{
+                //if looking at first image
+                Toast.makeText(getActivity().getBaseContext(), "This is the first image",
```

Comment:
You would want to put this in the strings.xml file in case your app needs to be translated or the UX people determine other phrases fit better.